net/trojan-go/Portfile

```diff
@@ -0,0 +1,274 @@
+                        lock    v0.5.0 \
+                        rmd160  f28cfe463c2aa119f5ea32b5373cdc06e606c3fd \
+                        sha256  4c228ad175fb924cd4c5ab08b281685f636ae28439e5508b3946964919b9c746 \
+                        size    98601 \
+                    github.com/golang/protobuf \
+                        lock    v1.4.2 \
+                        rmd160  fbf4477bc008421fde463d79f7bc54a36de91db2 \
+                        sha256  206d74f8fd066bb178135ee9c092e986f8a1e1104df242e148e99e5a839e4ef2 \
+                        size    171802 \
+                    github.com/golang/mock \
+                        lock    v1.2.0 \
+                        rmd160  5bb4b156a43c846427bd63c990bf89d8f3b9a2d7 \
+                        sha256  542c06c4b404f45569ebb4b2ddb976e6c55198c3e939bebf7def4922a78e671d \
+                        size    46504 \
+                    github.com/golang/glog \
+                        lock    23def4e6c14b \
+                        rmd160  b5bd9166cd1e073a035b5bbd3c4d9febf2c917a7 \
+                        sha256  2826d20759090e909ba0f8771def236ad6433fc3e44bdc28374b309efe3e57cf \
+                        size    19662 \
+                    github.com/go-sql-driver/mysql \
+                        lock    v1.5.0 \
+                        rmd160  c619fb55acd917f9b80c568f54b829836dd757e1 \
+                        sha256  cedc3d58292b89f2d5015dcfb6c7ab41c514cf9ce1b3733285743dc676ec8733 \
+                        size    90506 \
+                    github.com/davecgh/go-spew \
+                        lock    v1.1.0 \
+                        rmd160  0303eae19a01f38fe314921fd965e4d09b9ef3ad \
+                        sha256  810a597004388d68bb92d8aa612375419ba1080dd5fc2c66dd41b58f0ba4442c \
+                        size    42348 \
+                    github.com/aead/chacha20 \
+                        lock    8b13a72661da \
+                        rmd160  d1a095ea09c39b5a7366c44ed1715b12548f06bd \
+                        sha256  83aa07a56940cefc9f868bc48f1bba9ede6e5b314ea0a01ee662a7fc591c582d \
+                        size    21586 \
+                    github.com/riobard/go-bloom \
+                        lock    cdc8013cb5b3 \
+                        rmd160  516e59eed4200c50f09b4a4e32751cd59c37b02e \
+                        sha256  27c2479459dd154b496a019d595bd2863d47b98f9d9c9e9b7c97125636685987 \
+                        size    5502 \
+                    v2ray.com/core \
+                        repo    github.com/v2fly/v2ray-core \
+                        lock    v4.31.0 \
+                        rmd160  0f3ad31ab69415cee2f60a3918d206c8e8202d1a \
+                        sha256  8df67dac5968eec6c6beaa81884bbece5c1097d192c165698f8a559964a38b86 \
+                        size    7676992
+# Notes for updating this port:
+# Replace go.googlesource.com/protobuf with github.com/protocolbuffers/protobuf-go
+# github.com/riobard/go-bloom and v2ray.com/core need to be manually added
```

Comment:
Why do these need to be manually added?

Reply:
go2port output doesn’t have these two and the build fails so I had to add them in.

Reply:
Actually I think these are properly handled with `repo` entries. Just a second.

Reply:
No, it wasn't what I thought it was; it was a `replace` directive:

https://github.com/p4gefau1t/trojan-go/blob/d051cf4c8852d708769ca1c4e514306a88da830b/go.mod#L32

Instead of adding another `go.vendors` line, I would add a symlink in the post-extract phase like:

https://github.com/macports/macports-ports/blob/35318ba41f3d1865841ddd8303e5073eb7e4401a/editors/micro/Portfile#L42-L45

But I don't see where `github.com/riobard/go-bloom` is coming from; there don't seem to be any references in the upstream repo. I'll check it out.

Reply:
I have no idea why `github.com/riobard/go-bloom` isn't present in the dependency graph anywhere. I've never seen that before.

I pushed two changes:

1. Use symlink to handle the replace directive for `v2ray.com/core`
2. Use the correct version for `github.com/riobard/go-bloom`: this dependency comes from `github.com/shadowsocks/go-shadowsocks2` v0.1.0 which locks to `github.com/riobard/go-bloom` 218e1707c495 as you can see [here](https://github.com/shadowsocks/go-shadowsocks2/blob/b0fcb11abbfa3916ba57db3db18485a210f1a04a/go.mod#L6)